pkg/controller/pod_driver.go

```diff
@@ -133,26 +127,36 @@ func (p *PodDriver) checkAnnotations(ctx context.Context, pod *corev1.Pod) error
 	lock.Lock()
 	defer lock.Unlock()
 
-	annotation := make(map[string]string)
+	delAnnotations := []string{}
 	var existTargets int
 	for k, target := range pod.Annotations {
 		if k == util.GetReferenceKey(target) {
 			deleted, exists := p.mit.deletedPods[getPodUid(target)]
 			if deleted || !exists {
 				// target pod is deleted
+				delAnnotations = append(delAnnotations, k)
 				continue
 			}
 			existTargets++
 		}
-		annotation[k] = target
 	}
 
-	if existTargets != 0 {
-		delete(annotation, config.DeleteDelayAtKey)
+	if existTargets != 0 && pod.Annotations[config.DeleteDelayAtKey] != "" {
+		delAnnotations = append(delAnnotations, config.DeleteDelayAtKey)
 	}
-	if len(pod.Annotations) != len(annotation) {
-		if err := util.PatchPodAnnotation(ctx, p.Client, pod, annotation); err != nil {
-			klog.Errorf("Update pod %s error: %v", pod.Name, err)
+	if len(delAnnotations) != 0 {
+		// check mount pod resourceVersion, if it is not the latest, return conflict
+		newPod, err := p.Client.GetPod(ctx, pod.Name, pod.Namespace)
+		if err != nil {
+			return err
+		}
+		if newPod.ResourceVersion != pod.ResourceVersion {
```

Comment:
I think this version check is unnecessary, as the following `DelPodAnnotation` should return a conflict error if the pod is outdated

Reply:
patch api will not check resourceVersion, so we check it before patch.